### DISCUSSION TITLE
Does mongo-memory-server support sharded cluster？

### DISCUSSION THREAD

**Discussion by Jun-Ao** [typegoose/mongodb-memory-server#724] [Q&A]
## Versions

- NodeJS: 14.18.0
- mongodb-memory-server-*: 8.10.1 
- mongodb(the binary version): 4.0.21
- mongodb(the js package): 3.7.3
- mongoose: 5.13.13 
- system: Windows 10 (develop), linux (work)

package: mongo-memory-server 

## What is your question?

My project uses the sharded cluster mode of mongodb, I want to use mongo-memory-server to test the code, but I did not see the documentation that if you create a shard-cluster mongodb service, does mongo-memory-server support shard-cluster？

**Comment by hasezoey** [comment]
currently mongodb-memory-server only supports the binary `mongod`, the mongodb single instance and replset binary.

i dont know if sharding will ever be supported, because it would require not just adding another binary and handling, but also way more inter-op between already existing handling. (though it is not impossible, i just dont see a need for this)

keep in mind MMS is mainly meant for testing against a real database (not just mocked) with easy setup and handling and also "platform agnostic" (as long as nodejs(or any backend js engine) and mongodb support those platforms)

---

also, i dont know much about sharding, from my understanding it is just to better load-balance but the database at the end is the same, so i dont really see what this would bring if MMS would be able to handle it.

PS: i have moved this issue to a Q&A discussion, if you have a use-case that fits MMS goal, consider opening up a discussion explaining the use-case and what it would entail (if needed)
